r-package/grf/R/boosted_regression_forest.R

```diff
@@ -89,29 +89,29 @@
 #'
 #' @export
 boosted_regression_forest <- function(X, Y,
+                                      num.trees = 2000,
                                       sample.weights = NULL,
+                                      clusters = numeric(),
+                                      samples.per.cluster = 0,
                                       sample.fraction = 0.5,
-                                      mtry = NULL,
-                                      num.trees = 2000,
-                                      num.threads = NULL,
-                                      min.node.size = NULL,
+                                      mtry = min(ceiling(sqrt(ncol(X)) + 20), ncol(X)),
+                                      min.node.size = 5,
                                       honesty = TRUE,
-                                      honesty.fraction = NULL,
-                                      honesty.prune.leaves = NULL,
+                                      honesty.fraction = 0.5,
+                                      honesty.prune.leaves = TRUE,
+                                      alpha = 0.05,
+                                      imbalance.penalty = 0,
                                       ci.group.size = 2,
-                                      alpha = NULL,
-                                      imbalance.penalty = NULL,
-                                      seed = NULL,
-                                      clusters = NULL,
-                                      samples.per.cluster = NULL,
-                                      tune.parameters = FALSE,
-                                      num.fit.trees = 10,
-                                      num.fit.reps = 100,
-                                      num.optimize.reps = 1000,
+                                      tune.parameters = "none",
+                                      tune.num.trees = 10,
+                                      tune.num.reps = 100,
+                                      tune.num.draws = 1000,
                                       boost.steps = NULL,
                                       boost.error.reduction = 0.97,
                                       boost.max.steps = 5,
-                                      boost.trees.tune = 10) {
+                                      boost.trees.tune = 10,
+                                      num.threads = 0,
+                                      seed = runif(1, 0, .Machine$integer.max)) {
```

Comment:
Feels funny to pass a double as a seed. How about:
```r
seed = sample(.Machine$integer.max, 1)
```

Reply:
`runif(1, 0, .Machine$integer.max)` is always an integer, but surprisingly replacing it with `sample(.Machine$integer.max, 1)` caused some tests to fail (thresholds changed)